### PR TITLE
Feature/3 CI pipeline (resolves #3)

### LINKED ISSUE #3: Setup a CI pipeline
The pipeline should:
- [x] Run linting for source code (typescript, HTML, CSS)
- [x] Run format check for source code (typescript, HTML, CSS)
- [x] Run automated tests
- [x] Check for test coverage

### PR DESCRIPTION
**Pull request by Krismix1**
Changes introduced:
- Format checking using Prettier for HTML, TS, SCSS
- Couple modifications to Angular configuration
- Linting of SCSS using `stylelint`
- Configured `pre-commit` hook using `husky`
- Run tests in Firefox, and for CI run in Firefox and Chrome
- Add Github Action for CI - format, lint, and unit test

### CODE REVIEW

File: .editorconfig [hunk 1]
```diff
@@ -7,9 +7,8 @@ indent_style = space
 indent_size = 2
 insert_final_newline = true
 trim_trailing_whitespace = true
-
-[*.ts]
-quote_type = single
+end_of_line = lf
+max_line_length = 120
```
Comment (bambooSocks) [inline .editorconfig:11]: isn't 120 too long??? I think Prettier recommended 90

File: .prettierrc [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+  "tabWidth": 2,
+  "useTabs": false,
+  "semi": true,
+  "singleQuote": true,
+  "quoteProps": "as-needed",
+  "bracketSpacing": true,
+  "arrowParens": "always",
+  "htmlWhitespaceSensitivity": "css",
+  "endOfLine": "lf",
+  "printWidth": 120
```
Comment (bambooSocks) [inline .prettierrc:12]: same here

**Comment by bambooSocks** [comment]
also your tests don't seem to pass so have a look at that

File: .editorconfig (hunk 1 above)
Comment (Krismix1) [reply .editorconfig]: changed to 90

File: .prettierrc (hunk 2 above)
Comment (Krismix1) [reply .prettierrc]: changed to 90

**Comment by Krismix1** [comment]
> also your tests don't seem to pass so have a look at that

everything is passing now

**Pull request merged**